### PR TITLE
 EZP-28493: Interface improvements when assigning roles to user / groups

### PR DESCRIPTION
**Pull request by webhdx**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-28493
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Screenshot:
roles

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by lserwatka** [comment]
@micszo we need very careful QA on this one, including DB entires.

**Comment by lserwatka** [comment]
@webhdx could you attach screenshot?

File: src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
         ReactDOM.render(React.createElement(eZ.modules.UniversalDiscovery, {
-            onConfirm: onConfirm.bind(this, form),
-            onCancel: onCancel,
+            onConfirm: selectSubtreeConfirm.bind(this),
+            onCancel: closeUDW,
+            multiple: true,
             restInfo: {token, siteaccess}
         }), udwContainer);
     };
+    const toggleDisableState = () => {
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js]: `toggleDisabledState`

File: src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
+        limitationsRadio.forEach(radio => {
+            const disableNode = doc.querySelector(radio.dataset.disableSelector);
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js]: Do you have to query the whole document?

File: src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js [hunk 3]
```diff
@@ -1,25 +1,38 @@
...
+            if (disableNode) {
+                disableNode[methodName]('disabled', 'disabled');
+            }
+        });
+    };
 
-    btns.forEach(btn => btn.addEventListener('click', openUDW, false));
-})();
+    doc.querySelector('.ez-btn--select-subtree').addEventListener('click', selectSubtree, false);
+    limitationsRadio.forEach(radio => radio.addEventListener('change', toggleDisableState, false));
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js]: I'm not sure the use case here. If you're clicking on a specific radio button then I would expect to change the state of a clicked button. The `toggleDisabledState` function tries to toggle the state of all radio buttons. It looks quite suboptimal.

File: src/bundle/Resources/views/admin/role_assignment/create.html.twig [hunk 4]
```diff
@@ -38,10 +38,39 @@
                 <div class="ez-card__header ez-card__header--secondary">
                     {{ 'role_assignment.view.add.panel.limitations.title'|trans|desc('Limitations') }}
                 </div>
-                <div class="card-body">
-                    {{ form_row(form.sections) }}
-                    {{ form_widget(form.locations.select_content, {'attr': {'class': 'btn btn-secondary btn--open-udw'}}) }}
-                    {% do form.locations.setRendered %}
+                <div class="card-body ez-limitations">
+                    <div>
```
Comment (sunpietro) [inline src/bundle/Resources/views/admin/role_assignment/create.html.twig]: Do you need the `div` wrapper here?

File: src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js (hunk 3 above)
Comment (dew326) [reply src/bundle/Resources/public/js/scripts/admin.role_assignment.add.js]: The event change is fired only on radio which gets the checked=true. I have to loop over every to set disable state on other buttons

**Comment by lserwatka** [comment]
@mikadamczyk it looks like you need to run cs-fixer here

**Pull request merged**